<?xml version="1.0"?>
<robot xmlns:xacro="http://ros.org/wiki/xacro">

  <xacro:property name="laser_min_range" value="0.1"/>
  <xacro:property name="laser_max_range" value="200.0"/>
  <xacro:property name="horizontal_fov" value="81.7"/>
  <xacro:property name="vertical_fov" value="25.1"/>
  <xacro:property name="ros_topic" value="livox/lidar"/>
  <xacro:property name="samples" value="24000"/>
  <xacro:property name="publish_pointcloud_type" value="0"/>
  <xacro:property name="csv_file_name" value="package://livox_laser_simulation/scan_mode/horizon.csv"/>

  <xacro:include filename="$(find livox_laser_simulation)/urdf/livox_gazebo_sensor.xacro"/>
  <xacro:macro name="Livox_Horizon" params="visualize:=True name:=livox publish_pointcloud_type:=0
                                            ros_topic:=livox/lidar downsample:=1
                                            noise_mean:=0.0 noise_stddev:=0.01 imu_noise_stddev:=0.00264">
    <link name="${name}_base">
      <xacro:null_inertial/>
      <visual>
        <geometry>
          <mesh filename="package://livox_laser_simulation/meshes/livox_mid40.dae"/>
        </geometry>
      </visual>

      <collision>
      <geometry>
        <mesh filename="package://livox_laser_simulation/meshes/livox_mid40.dae"/>
      </geometry>
      </collision>
    </link>
    <link name="${name}">
      <xacro:null_inertial/>
    </link>

    <joint name="${name}_to_${name}_base_joint" type="fixed">
      <parent link="${name}_base"/>
      <child link="${name}"/>
    </joint>

    <link name="${name}_imu_link"/>

    <joint name="imu_link_joint" type="fixed">
      <parent link="${name}"/>
      <child link="${name}_imu_link"/>
      <origin xyz="-0.05512 -0.02226 0.0297"/>
    </joint>

    <xacro:Livox_gazebo_sensor name="${name}" visualize="${visualize}" publish_pointcloud_type="${publish_pointcloud_type}"
                               noise_mean="${noise_mean}" noise_stddev="${noise_stddev}" imu_noise_stddev="${imu_noise_stddev}"/>
  </xacro:macro>
</robot>
